src/types.rs

```diff
@@ -140,12 +140,18 @@ impl ContractResult {
     }
 }
 
+#[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, JsonSchema)]
+pub struct LogAttribute {
+    pub key: String,
+    pub value: String,
+}
+
 #[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, JsonSchema)]
 pub struct Response {
     // let's make the positive case a struct, it contrains Msg: {...}, but also Data, Log, maybe later Events, etc.
     pub messages: Vec<CosmosMsg>,
-    pub log: Option<String>,  // abci defines this as string
-    pub data: Option<Binary>, // abci defines this as bytes
+    pub log: Option<Vec<LogAttribute>>, // abci defines this as string
```

Comment:
Can we make this non-optional, like in `messsage`? This would avoid two different ways to express _no logs_

Reply:
Sure. I started to make them Optional, as we actually need that for anything we may accept as input (cuz go serializes `[]` as `null` and we must use `Option<Vec<T>>` to parse that. However, this is only a return value. Yes, good adjustment,.